ostelco-ios-client/Coordinators/EKYC/SingaporeEKYCCoordinator.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  SingaporeEKYCCoordinator.swift
+//  ostelco-ios-client
+//
+//  Created by Ellen Shapiro on 6/4/19.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import ostelco_core
+import UIKit
+
+class SingaporeEKYCCoordinator: EKYCCoordinator {
+    enum Destination {
+        case goBackAndChooseCountry
+        case landing
+        case selectVerificationMethod
+        case jumio
+        case singPass
+        case editSingPassAddress(address: MyInfoAddress?, delegate: MyInfoAddressUpdateDelegate)
+        case verifySingPassAddress(queryItems: [URLQueryItem])
+        case enterAddress
+        case stepsForNRIC
+        case enterNRIC
+        case ekycRejected
+        case success(region: RegionResponse)
+        case waitingForVerification
+    }
+    
+    let navigationController: UINavigationController
+    
+    weak var delegate: EKYCCoordinatorDelegate?
+    var jumioCoordinator: JumioCoordinator?
+    var singPassCoordinator: SingPassCoordinator?
+    let country = Country("sg")
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+    }
+    
+    func determineAndNavigateDestination(from region: RegionResponse?, animated: Bool) {
+        let destination = self.determineDestination(from: region)
+        self.navigate(to: destination, animated: animated)
+    }
+    
+    func determineDestination(from region: RegionResponse?) -> SingaporeEKYCCoordinator.Destination {
+        guard let region = region else {
+            return .landing
+        }
+        
+        switch region.status {
+        case .APPROVED:
+            return .success(region: region)
+        case .REJECTED:
+            return .ekycRejected
+        case .PENDING:
+            guard
+                let jumio = region.kycStatusMap.JUMIO,
+                let addressAndPhoneNumber = region.kycStatusMap.ADDRESS_AND_PHONE_NUMBER,
+                let nricFin = region.kycStatusMap.NRIC_FIN,
+                let myInfo = region.kycStatusMap.MY_INFO else {
+                    return .goBackAndChooseCountry
+            }
+            
+            switch (jumio, addressAndPhoneNumber, nricFin, myInfo) {
```

Comment:
would appreciate as many extra sets of eyes on this bit of logic as possible @havard024 @prasanthu

Reply:
Is there a diagram of this somewhere? Seems like important business logic stuff.

Reply:
Not that I've seen but maybe @havard024 @prasanthu know?